### ISSUE TITLE
Request to add feature setting match file type by using regex. 

### ISSUE THREAD

**Issue by hoangnx30** [zed-industries/zed#10765]
### Check for existing issues

- [X] Completed

### Describe the feature

Currently, I saw file_types only matches word by word. Can team support to match by regex? 
```json
"file_types": {
    "Dockerfile": [
      "Dockerfile",
      "Dockerfile.dev",
      "Dockerfile.prod",
      "Dockerfile.uat"
    ]
  }
``` 
------> It worked.

```json
"file_types": {
    "Dockerfile": [
      "Dockerfile",
      "Dockerfile.*",
    ]
  }
```

-----> it didn't work

**Comment by osiewicz** [comment]
I think a better alternative would be to support matching via globs, not regexes; https://docs.rs/glob/latest/glob/ It's stricter than a regex, but it should suffice for your use case.

**Comment by guibes** [comment]
Nice, anyone can help me, will be my first issue, I can try fix it.

**Comment by osiewicz** [comment]
Sure! For starters, this function does path filtering: https://github.com/zed-industries/zed/blob/74241d9f935158dc113040c3d2c8b3785fdf0f09/crates/language/src/language_registry.rs#L472
`user_file_types` is a deserialized content of `file_types`, e.g. for the example you've provided:
```
"file_types": {
    "Dockerfile": [
      "Dockerfile",
      "Dockerfile.dev",
      "Dockerfile.prod",
      "Dockerfile.uat"
    ]
  }
```
`user_file_types` is gonna be `Some({"Dockerfile": [ "Dockerfile", "Dockerfile.dev", "Dockerfile.prod", "Dockerfile.uat" ]})`.

It then follows that all you have to do is change the code within the call to `any` to use globs instead: https://github.com/zed-industries/zed/blob/74241d9f935158dc113040c3d2c8b3785fdf0f09/crates/language/src/language_registry.rs#L488

**Comment by nathansbradshaw** [comment]
I believe the issue I am running into with my angular LSP is similar to this one. From what I gather, the Angular LSP should only be applied to HTML files that have a `component` prefix to the suffix IE `app.component.html` I made a [branch](https://github.com/zed-industries/zed/commit/291d8b5062d4369760a279f3a996793393022b3b) that fixed this issue, but reading here it looks like it's not the desired result. However, pulling down PR #10960 I am not seeing how I can get the LSP to recognize angular files.

**Comment by farayolaj** [comment]
I made PR #12043 to work on this. I picked up the work from where @guibes stopped, taking note of the reviews from @osiewicz.

**Comment by osiewicz** [comment]
This is gonna land in next Preview on Wednesday, 22.05. Thanks for getting this across the line @farayolaj :)

**Comment by JosephTLyons** [comment]
This is now available in [v0.136.0-pre](https://zed.dev/releases/preview/0.136.0).

**Comment by znycheporuk** [comment]
is it possible to show docker image near file that is considered dockerfile?
<img width="213" alt="image" src="https://github.com/zed-industries/zed/assets/42861377/05821a59-8e07-4b86-93fe-ffa94a9c8f4c">
